### PR TITLE
Staking ingame

### PR DESCRIPTION
**Pull request by tyrosine1153**
### Description

- Add staking button in Menu, Staking Popup, Staking Popup(staking step 0)

- Add StakeRegularFixedRewardSheet

- Add L10n keys
  - UI_MAIN_MENU_STAKING : "Staking"
  - UI_STAKING_POPUP_DESCRIPTION : "When you deposit NCG, the monsters go on an expedition to get the treasure."
  - UI_START_STAKING_UNIT : "Start staking minimum unit :"
  - UI_GO_TO_UPLOAD : "Go to upload"
  - UI_REWARD_TYPE : "Reward Type"

### Related Links

[몬스터 콜렉션 Ingame 구현](https://app.asana.com/0/958521740385861/1202312326927235/f)

### Screenshot
image
image
image

### CODE REVIEW

File: nekoyume/Assets/StreamingAssets/Localization/common.csv [hunk 1]
```diff
@@ -634,6 +634,7 @@ UI_MAIN_MENU_COMBINATION,Workshop,공방,Oficina,,ワークショップ,Werkstat
 UI_MAIN_MENU_QUEST,Stage!,스테이지!,Fase!,,ステージ！,Ebene!,,ด่าน!,Poziom!,,,Stage!,Niveau !,,,,层级！,
 UI_MAIN_MENU_RANKING,Arena,아레나,Arena,,アリーナ,Arena,Arena,อารีน่า,Arena,Arena,Arena,Arena,Arène,Арена,សៃវ្យេន,,竞技场,
 UI_MAIN_MENU_SHOP,Shop,상점,Comprar,,ショップ,Geschäft,Tienda,ร้านค้า,Sklep,Parduotuvė,Winkel,Toko,Boutique,Магазин,ហាង,,店铺,
+UI_MAIN_MENU_STAKING,Staking,스테이킹,,,,,,,,,,,,,,,,
```
Comment (sonohoshi) [inline nekoyume/Assets/StreamingAssets/Localization/common.csv:637]: I think this lable overlap [`UI_PLACE_STAKING`](https://github.com/planetarium/NineChronicles/blob/7d540098cd6674fcb8a82518c02460f22cf86d10/nekoyume/Assets/StreamingAssets/Localization/common.csv#L1059).

File: nekoyume/Assets/_Scripts/UI/Module/Item/StakingItemView.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            _disposables.DisposeAllAndClear();
+            baseItemView.Container.SetActive(true);
+            baseItemView.TouchHandler.gameObject.SetActive(true);
+            baseItemView.ItemImage.sprite = baseItemView.GetItemIcon(itemBase);
+            baseItemView.CountText.text = count.ToString();
+
+            baseItemView.TouchHandler.OnClick
+                .Select(_ => itemBase).Subscribe(onClick).AddTo(_disposables);
+            baseItemView.TouchHandler.OnExit
+                .Subscribe(_ => Debug.LogError("OnExit")).AddTo(_disposables);
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Module/Item/StakingItemView.cs:32]: Is this `Debug.LogError("OnExit")` need?

File: nekoyume/Assets/_Scripts/UI/Widget/Menu.cs [hunk 3]
```diff
@@ -186,6 +193,8 @@ var firstOpenMimisbrunnrKey
             mimisbrunnrExclamationMark.gameObject.SetActive(
                 btnMimisbrunnr.IsUnlocked
                 && PlayerPrefs.GetInt(firstOpenMimisbrunnrKey, 0) == 0);
+            
+            stakingLevelIcon.sprite = SpriteHelper.GetStakingIcon(States.Instance.StakingLevel);
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Menu.cs:197]: It is good too, however we have `MonsterCollectionStateSubject`.
This class has reactive-stream about `StakingLevel`. You can subscribe this stream like it...
```cs
MonsterCollectionStateSubject.Level.Subscribe(level => 
    stakingLevelIcon.sprite = SpriteHelper.GetStakingIcon(level))
    .AddTo(gameObject);
```

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+        private const string StepScoreFormat = "{0}<size=18><color=#A36F56>/{1}</color></size>";
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            stakingButton.OnSubmitSubject.Subscribe(_ =>
+            {
+                // Do Something
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:43]: Is this code for the unfinished function?

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            {
+                Close();
+                AudioController.PlayClick();
+            });
+
+            tooltip.SetActive(false);
+
+            this.UpdateAsObservable()
+                .Where(_ => tooltip.activeSelf && Input.GetMouseButtonDown(0))
+                .Subscribe(_ => tooltip.SetActive(false));
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:57]: Does not need `Dispose()`?

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+                    var result = deposit / regular.Rewards[i].Rate;
+                    var levelBonus = regularFixed.Rewards.FirstOrDefault(
+                        reward => reward.ItemId == regular.Rewards[i].ItemId)?.Count ?? 0;
+                    result += levelBonus;
+
+                    var view = regularItemViews[i];
+                    view.Set(item, result, itemBase =>
+                    {
+                        ShowToolTip(itemBase, view.gameObject.transform); // 틀림
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:87]: This comment is intended?

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+        [SerializeField] private TextMeshProUGUI stepText;
+
+        [SerializeField] private TextMeshProUGUI stepScoreText;
+
+        [SerializeField] private StakingItemView[] regularItemViews;
+
+        [SerializeField] private GameObject tooltip;
+
+        [SerializeField] private RectTransform tooltipRectTransform;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:31]: Are `GameObject tooltip` and `RectTransform tooltipRectTransform` component of diffrent objects?
If it is same... can remove `GameObject tooltip`.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            this.UpdateAsObservable()
+                .Where(_ => tooltip.activeSelf && Input.GetMouseButtonDown(0))
+                .Subscribe(_ => tooltip.SetActive(false));
+        }
+
+        public override void Show(bool ignoreStartAnimation = false)
+        {
+            var level = States.Instance.StakingLevel;
+            var deposit = States.Instance.StakedBalanceState?.Gold.MajorUnit ?? 0;
+            var regularSheet = Game.Game.instance.TableSheets.StakeRegularRewardSheet;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:64]: `Game.Game.instance.TableSheets` is same with `TableSheets.Instance`.

**Review by sonohoshi — CHANGES_REQUESTED** [review]

File: nekoyume/Assets/StreamingAssets/Localization/common.csv (hunk 1 above)
Comment (tyrosine1153) [reply nekoyume/Assets/StreamingAssets/Localization/common.csv:637]: I will replace with your suggestion. but I have a question, does the existing name of monster collecting stay the same?

File: nekoyume/Assets/_Scripts/UI/Module/Item/StakingItemView.cs (hunk 2 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Module/Item/StakingItemView.cs]: This is an unneeded comment. I deleted it.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs (hunk 4 above)
Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:42]: There is no need to erase, but it seems need writing down the annotation in more detail.
ex) This button not works yet.

File: nekoyume/Assets/_Scripts/UI/Widget/Menu.cs (hunk 3 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Menu.cs]: This is awesome... Thank you!

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs (hunk 6 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs]: Not intended. I fixed it, Thank you!

**Review by sonohoshi — APPROVED** [review]
LGTM 👏

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs (hunk 7 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs:30]: I applied this, Thank you!

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs (hunk 8 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/StakingPopup.cs]: I applied this, Thank you!

**Pull request merged**